src/components/FindMovie/FindMovie.tsx

```diff
@@ -17,41 +68,64 @@ export const FindMovie: React.FC = () => {
               id="movie-title"
               placeholder="Enter a title to search"
               className="input is-dander"
+              value={query}
+              onChange={event => {
+                setQuery(event.target.value);
+                setHasNoSuchTitle(false);
+              }}
             />
           </div>
 
-          <p className="help is-danger" data-cy="errorMessage">
-            Can&apos;t find a movie with such a title
-          </p>
+          {hasNoSuchTitle && (
+            <p className="help is-danger" data-cy="errorMessage">
+              Can&apos;t find a movie with such a title
+            </p>
+          )}
         </div>
 
         <div className="field is-grouped">
           <div className="control">
             <button
               data-cy="searchButton"
               type="submit"
-              className="button is-light"
+              className={classNames(
+                'button',
+                'is-light',
+                { 'is-loading': isLoading },
+              )}
+              disabled={!query}
             >
-              Find a movie
+              {!movieFromServer
+                ? 'Find a movie'
+                : 'Search again'}
             </button>
           </div>
 
           <div className="control">
-            <button
-              data-cy="addButton"
-              type="button"
-              className="button is-primary"
-            >
-              Add to the list
-            </button>
+            {movieFromServer && (
+              <button
+                data-cy="addButton"
+                type="button"
+                className="button is-primary"
+                onClick={() => {
+                  onAddMovie(movieFromServer);
+                  setQuery('');
+                  setMovieFromServer(null);
```

Comment:
also move this logic to separate function 
and call here onClick={addingMovie} as example
const addingMovie = () => {
onAddMovie(movieFromServer);
setQuery('');
setMovieFromServer(null);
}